OpenRA.Mods.Cnc/Traits/World/WithResourceAnimation.cs

```diff
@@ -0,0 +1,116 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2020 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Graphics;
+using OpenRA.Mods.Common;
+using OpenRA.Mods.Common.Effects;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Cnc.Traits
+{
+	[Desc("Allows to play animations on resources.", "Attach this to the world actor.")]
+	public class WithResourceAnimationInfo : TraitInfo, Requires<IResourceLayerInfo>
+	{
+		[FieldLoader.Require]
+		[Desc("Resource types to animate.")]
+		public readonly HashSet<string> Types = null;
+
+		[Desc("The percentage of resource cells to play the animation on.", "Use two values to randomize between them.")]
+		public readonly int[] Ratio = { 5 };
+
+		[Desc("Tick interval between two animation spawning.", "Use two values to randomize between them.")]
+		public readonly int[] Interval = { 150 };
+
+		[FieldLoader.Require]
+		[Desc("Animation image.")]
+		public readonly string Image = null;
+
+		[SequenceReference(nameof(Image))]
+		[Desc("Animation sequences.")]
+		public readonly string[] Sequences = new string[] { "idle" };
```

Comment:
`"Randomly select one of these sequences to render"` or similar, telling the docs reader how the provided values are used.

Reply:
Thanks. That makes the documentation line actually useful.